lib/main.coffee

```diff
@@ -48,7 +48,9 @@ module.exports =
         parameters.push('--define', 'display_errors=On')
         parameters.push('--define', 'log_errors=Off')
         text = textEditor.getText()
-        return helpers.exec(command, parameters, {stdin: text}).then (output) ->
+        [projectPath] = atom.project.relativizePath(filePath)
+        cwd = if projectPath? then projectPath else filePath.replace(/\\/g, '/').replace(/\/[^\/]*\/?$/, '')
```

Comment:
Is that mess of regex attempting to get the directory of the path? Why aren't you just using `path.dirname(filePath)`?